src/android/FileUtils.java

```diff
@@ -1124,7 +1171,18 @@ public void run(JSONArray args) throws FileExistsException, IOException, TypeMis
                     }
                 }, lastRawArgs, callback);
                 break;
-            case WRITE_PERM:
+            case PERM_DIR:
+                threadhelper( new FileOp( ){
+                    public void run(JSONArray args) throws FileExistsException, IOException, TypeMismatchException, EncodingException, JSONException {
+                        String dirname = args.getString(0);
+
+                        String path = args.getString(1);
+                        JSONObject obj = getFile(dirname, path, args.optJSONObject(2), true);
+                        callback.success(obj);
+                    }
+                }, lastRawArgs, callback);
+                break;    
+            case WRITE_ACTION:
```

Comment:
I like this convention better since this differentiates the calls for 'FILE' and 'DIR'.